tests/Feature/Api/V1/Barcodes/ScanBarcodeTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Tests\Feature\Api\V1\Barcodes;
+
+use App\Api\V1\Items\Resources\ItemResource;
+use App\Api\V1\Orders\Resources\OrderResource;
+use App\Api\V1\Processes\Resources\ProcessResource;
+use App\Api\V1\StorageLocations\Resources\StorageLocationResource;
+use App\Api\V1\Users\Resources\StaffMemberResource;
+use App\Api\V1\Warehouses\Resources\WarehouseResource;
+use App\Api\V1\Warehouses\Resources\WorkstationResource;
+use Domain\Items\Models\Item;
+use Domain\Orders\Models\Order;
+use Domain\Processes\Models\Process;
+use Domain\Users\Models\StaffMember;
+use Domain\Warehouses\Models\StorageLocation;
+use Domain\Warehouses\Models\Warehouse;
+use Domain\Warehouses\Models\Workstation;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Testing\Fluent\AssertableJson;
+
+uses(RefreshDatabase::class);
+
+it('returns the correct resource when a barcode is scanned', function (Collection $scannedEntities, $type, $resource) {
+    foreach ($scannedEntities as $scannedEntity) {
+        $response = $this
+            ->getJson(route('barcodes.scan', ['barcode' => $scannedEntity->barcode->barcode]))
+            ->assertOk()
+            ->assertJson(fn (AssertableJson $json) => $json
+                ->where('data.type', $type)
```

Comment:
```suggestion
it('returns the correct resource when a barcode is scanned', function (Collection $scannedEntities, $resource) {
    foreach ($scannedEntities as $scannedEntity) {
        $type = str_replace('Resource', '', last(explode('\\', $resource)));
    
        $response = $this
            ->getJson(route('barcodes.scan', ['barcode' => $scannedEntity->barcode->barcode]))
            ->assertOk()
            ->assertJson(fn (AssertableJson $json) => $json
                ->where('data.type', $type)
```
non-blocking: could be made dynamic like so or use the first entity, get_class on it then explode etc.